addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6623,6 +6623,15 @@ registry.ImageTools = ImageHandlerOption.extend({
         return this._super(...arguments);
     },
 
+    /**
+     * @override
+     */
+    selectAttribute(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === "alt" && params.activeValue.trim() !== "") {
+            this.$target[0].removeAttribute("role");
```

Comment:
An image can have an `role` different from `presentation` or `none`. We do not necessarily have to remove the `role` attribute of an image that has an `alt !== ""`, e.g., `<img src="submit-icon.png" alt="Submit the form" role="button">`